src/main/crab_verifier_timed.cpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Prevail Verifier contributors.
+// SPDX-License-Identifier: MIT
+#include "crab_verifier.hpp"
+
+template<typename F>
+auto timed_execution(F f) {
+    clock_t begin = clock();
+
+    const auto& res = f();
+
+    clock_t end = clock();
+
+    double elapsed_secs = double(end - begin) / CLOCKS_PER_SEC;
+    return std::make_tuple(res, elapsed_secs);
+}
+
+/// First item of returned value is true if the program passes verification
+std::tuple<bool, double> run_ebpf_analysis_timed(cfg_t& cfg, program_info info, ebpf_verifier_options_t* options) {
```

Comment:
I'm not sure about this function, with its dedicated .h file and declaration. It can be inlined into the (single) calling code, and `timed_execution()` can go to some generic `utils.h` file.

Reply:
Good idea, done